### ISSUE TITLE
An overload with a parameter of type `object` is not used

### ISSUE THREAD

**Issue by lostmsu** [pythonnet/pythonnet#811]
### Environment

-   Pythonnet version: 2.3.0
-   Python version: 3.6
-   Operating System: Win 10 x64 1809

### Details

I am trying to get Python call my C# method, which has a parameter of type `object` with an instance of Python's `KeyError`.

```csharp
var o = overloaded.ToPython();

dynamic callWithSelf = PythonEngine.Eval("lambda o: o.ObjOrClass(KeyError())");
callWithSelf(o);
```

Where `ObjOrClass` definition is: `public void ObjOrClass(object _) => this.Value = Object;`. The actual function body here is not important.

-  Error:

```
Python.Runtime.PythonException : TypeError : No method matches given arguments for ObjOrClass
```

### Possibly related:
https://github.com/pythonnet/pythonnet/issues/265
https://github.com/pythonnet/pythonnet/issues/782

### Thoughs
IMHO, Python.NET should create a `PyObject` and pass it here.

**Comment by filmor** [comment]
Yes, it's a possibility to do that. The issue at hand is probably, that Python objects do not derive from `System.Object` and can thus not match the overload. Can you prepare a PR?

**Comment by lostmsu** [comment]
@filmor I am not sure I understood your comment. You can see my fix in the commit e9712b4 above. This change is potentially backwards-incompatible in case there's a .NET method with two overloads, one having parameter type `object`, and another `PyObject` for some parameter.

**Comment by filmor** [comment]
I'll need to look into this, could you make a PR from this commit?
